PurchasesCoreSwift/Networking/Backend.swift

```diff
@@ -0,0 +1,614 @@
+                     productIdentifier: String,
+                     subscriptionGroup: String,
+                     receiptData: Data,
+                     appUserID: String,
+                     completion: @escaping OfferSigningResponseHandler) {
+    // swiftlint:enable function_parameter_count
+        let fetchToken = receiptData.base64EncodedString(options: .lineLength64Characters)
+
+        let requestBody: [String: Any] = ["app_user_id": appUserID,
+                                          "fetch_token": fetchToken,
+                                          "generate_offers": [
+                                            ["offer_id": offerIdentifier,
+                                             "product_id": productIdentifier,
+                                             "subscription_group": subscriptionGroup
+                                            ]
+                                          ]]
+
+        self.httpClient.performPOSTRequest(serially: true, path: "/offers",
+                                           requestBody: requestBody,
+                                           headers: authHeaders) { statusCode, response, error in
+            if let error = error {
+                completion(nil, nil, nil, nil, ErrorUtils.networkError(withUnderlyingError: error))
+                return
+            }
+
+            guard statusCode < HTTPStatusCodes.redirect.rawValue else {
+                let code = self.maybeNumberFromError(code: response?["code"])
+                let backendMessage = response?["message"] as? String
+                let error = ErrorUtils.backendError(withBackendCode: code, backendMessage: backendMessage)
+                completion(nil, nil, nil, nil, error)
+                return
+            }
+
+            guard let offers = response?["offers"] as? [[String: AnyObject]], offers.count > 0 else {
+                completion(nil, nil, nil, nil, ErrorUtils.unexpectedBackendResponseError())
+                return
+            }
+
+            let offer = offers[0]
+            if let signatureError = offer["signature_error"] as? [String: AnyObject] {
+                let code = self.maybeNumberFromError(code: signatureError["code"])
+                let backendMessage = signatureError["message"] as? String
+                let error = ErrorUtils.backendError(withBackendCode: code, backendMessage: backendMessage)
+                completion(nil, nil, nil, nil, error)
+
+            } else if let signatureData = offer["signature_data"] as? [String: AnyObject] {
+                let signature = signatureData["signature"] as? String
+                let keyIdentifier = offer["key_id"] as? String
+                let nonceString = signatureData["nonce"] as? String
+                let nonce: UUID?
+                if let nonceString = nonceString {
+                    nonce = UUID(uuidString: nonceString)
+                } else {
```

Comment:
[totally minor] Is this `else` block needed?

Reply:
Unfortunately, yeah, the `let nonce: UUID?` on line 200 isn't set yet, so it won't compile we don't explicitly set it.

Reply:
oohh, you're right,  If it was a `var`, it would be possible. I didn't realize that `nonce` is a constant 🤦🏻‍♂️